src/libasr/modfile.cpp

```diff
@@ -79,4 +107,27 @@ ASR::TranslationUnit_t* load_modfile(Allocator &al, const std::string &s,
     return tu;
 }
 
+ASR::TranslationUnit_t* load_pycfile(Allocator &al, const std::string &s,
+        bool load_symtab_id) {
+#ifdef WITH_LFORTRAN_BINARY_MODFILES
+    BinaryReader b(s);
+#else
+    TextReader b(s);
+#endif
+
+    std::string file_type = b.read_string();
+    if (file_type != lfortran_modfile_type_string) {
+        throw LCompilersException("LFortran Modfile format not recognized");
+    }
+    std::string version = b.read_string();
+    if (version != LFORTRAN_VERSION) {
+        throw LCompilersException("Incompatible format: LFortran Modfile was generated using version '" + version + "', but current LFortran version is '" + LFORTRAN_VERSION + "'");
+    }
+    std::string asr_binary = b.read_string();
+    ASR::asr_t *asr = deserialize_asr(al, asr_binary, load_symtab_id);
+
+    ASR::TranslationUnit_t *tu = ASR::down_cast2<ASR::TranslationUnit_t>(asr);
+    return tu;
+}
```

Comment:
Why not merging this with LFortran's save/load? It seems it is exactly the same code.

Reply:
Yeah. Well I would prefer to do in a separate PR. Reason - Don' want to break something in LFortran which already works. Also, for now this is exactly similar but we don't know if it will diverge in future. Other than similarity is there any other thing that is of concern?